docs/blog/2023-11-20-life-of-a-sample-part-1.md

```diff
@@ -0,0 +1,209 @@
+* When only the hashring is set, it acts as a **Router** and forwards the data to the correct ingestor.
+
+#### Handling Out-of-Order Timestamps
+
+To enhance reliability in data ingestion, Thanos Receive supports out-of-order samples.
+
+Samples are ingested into the Receiver's TSDB, which has strict requirements for the order of timestamps:
+
+* Samples are expected to have increasing timestamps for a given time series.
+* A new sample cannot be more than 1 hour older than the most recent sample of any time series in the TSDB.
+
+When these requirements are not met, the samples are dropped, and an out-of-order warning is logged. However, there are scenarios where out-of-order samples may occur, often because of [clients' misconfigurations](https://thanos.io/tip/operating/troubleshooting.md/#possible-cause-1) or delayed remote write requests, which can cause samples to arrive out of order depending on the remote write implementation. Additional examples at the Prometheus level can be found in [this article](https://promlabs.com/blog/2022/12/15/understanding-duplicate-samples-and-out-of-order-timestamp-errors-in-prometheus/).
+
+As you are not necessarily in control of your clients' setups, you may want to increase resilience against these issues. Support for out-of-order samples has been implemented for the TSDB. This feature can be enabled with the `tsdb.out-of-order.time-window` flag on the Receiver. The downsides are: 
+
+* An increase in the TSDB's memory usage, proportional to the number of out-of-order samples. 
+* The TSDB will produce blocks with overlapping time periods, which the compactor must handle. Ensure the `--compact.enable-vertical-compaction` [flag](https://thanos.io/tip/components/compact.md/#enabling-vertical-compaction) is enabled on the compactor to manage these overlapping blocks. We will cover this in more detail in the next article.
+
+Additionally, consider setting the `tsdb.too-far-in-future.time-window` flag to a value higher than the default 0s to account for possible clock drifts between clients and the Receiver.
+
+### Conclusion
+
+In this first part, we have covered the initial steps of the sample lifecycle in Thanos, focusing on the ingestion process. We have explored the remote write protocol, the Receive component, and the critical configurations needed to ensure high availability and durability. Now, our sample is safely ingested and stored in the system. In the next part, we will continue following our sample's journey, delving into the data management and querying processes.
+
+See the full list of articles in this series:
+
+* [Life of a sample in thanos, and how to configure it – Ingestion – Part I](2023-11-20-life-of-a-sample-part-1.md)
+* [Life of a sample in thanos, and how to configure it – Data Management – Part II](2023-11-20-life-of-a-sample-part-2.md)
+* [Life of a sample in thanos, and how to configure it – Querying – Part III](2023-11-20-life-of-a-sample-part-3.md)
+
```

Comment:
Those are going to fail CI. Better to remove links for now and mention that they will come.

Reply:
Ack, removed them.